benchmarks/src/main/scala/org/typelevel/otel4s/benchmarks/AttributesBenchmark.scala

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.benchmarks
+
+import io.opentelemetry.api.common.{AttributeKey => JAttributeKey}
+import io.opentelemetry.api.common.{Attributes => JAttributes}
+import org.openjdk.jmh.annotations._
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.AttributeKey
+import org.typelevel.otel4s.sdk.Attributes
+
+import java.util.concurrent.TimeUnit
+
+// benchmarks/Jmh/run org.typelevel.otel4s.benchmarks.AttributesBenchmark -prof gc
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(1)
+@Measurement(iterations = 15, time = 1)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Warmup(iterations = 5, time = 1)
+class AttributesBenchmark {
+  import AttributesBenchmark._
+
+  @Benchmark
+  def java_computeHashCode(): Unit = {
+    for (a <- jAttributes) { a.hashCode() }
+  }
+
+  @Benchmark
+  def java_ofOne(): JAttributes =
+    JAttributes.of(jKeys.head, values.head)
+
+  @Benchmark
+  def java_ofFive(): JAttributes =
+    JAttributes.of(
+      jKeys(0),
+      values(0),
+      jKeys(1),
+      values(1),
+      jKeys(2),
+      values(2),
+      jKeys(3),
+      values(3),
+      jKeys(4),
+      values(4)
+    )
+
+  @Benchmark
+  def java_builderTenItem(): JAttributes = {
+    val builder = JAttributes.builder()
+    for (i <- 0 until 10) {
+      builder.put(jKeys(i), values(i))
+    }
+    builder.build()
+  }
+
+  @Benchmark
+  def otel4s_computeHashCode(): Unit = {
+    for (a <- attributes) {
+      a.hashCode()
+    }
+  }
+
+  @Benchmark
+  def otel4s_ofOne(): Attributes =
+    Attributes(Attribute(keys.head, values.head))
+
+  @Benchmark
+  def otel4s_ofFive(): Attributes =
+    Attributes(
+      Attribute(keys(0), values(0)),
+      Attribute(keys(1), values(1)),
+      Attribute(keys(2), values(2)),
+      Attribute(keys(3), values(3)),
+      Attribute(keys(4), values(4))
+    )
+
+  @Benchmark
+  def otel4s_builderTenItem(): Attributes = {
+    val builder = Attributes.newBuilder
+    for (i <- 0 until 10) {
+      builder.addOne(keys(i), values(i))
+    }
+    builder.result()
+  }
+}
+
+object AttributesBenchmark {
+  private val values = List.tabulate(10)(i => s"value$i")
+
+  private val jKeys = List.tabulate(10)(i => JAttributeKey.stringKey(s"key$i"))
+  private val jAttributes = List.tabulate(10) { i =>
+    val size = i + 1
+    val pairs = jKeys.take(size).zip(values.take(size))
+
+    pairs
+      .foldLeft(JAttributes.builder()) { case (builder, (key, value)) =>
+        builder.put(key, value)
+      }
+      .build()
+  }
+
+  private val keys = List.tabulate(10)(i => AttributeKey.string(s"key$i"))
+  private val attributes = List.tabulate(10) { i =>
+    val size = i + 1
+    val pairs = keys.take(size).zip(values.take(size))
+
+    pairs
+      .foldLeft(Attributes.newBuilder) { case (builder, (key, value)) =>
+        builder.addOne(key, value)
+      }
+      .result()
+  }
```

Comment:
it's been so long since I've done any benchmarking, but I think we want these defined in the benchmark instance rather than the companion, and I think there's also a setup method where you're supposed to construct them to prevent some compiler optimisations that wouldn't happen in regular code

Reply:
I took the idea of the benchmark from https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/jmh/java/io/opentelemetry/api/common/AttributesBenchmark.java#L25-L40. 
Here, they are defined as static variables.